example/spec/distributionAPI/distributionSpec.js

```diff
@@ -76,51 +97,109 @@ describe('Distribution API', () => {
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      });
 
-    it('redirecting to Earthdata login for unauthorized requests to /s3credentials endpoint.', async () => {
-      const response = await got(
-        `${process.env.DISTRIBUTION_ENDPOINT}/s3credentials`,
-        { followRedirect: false }
-      );
-      const authorizeUrl = new URL(response.headers.location);
-      expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
-      expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
-      expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      it('redirects to Earthdata login for requests on /s3credentials endpoint.', async () => {
+        const response = await got(
+          `${process.env.DISTRIBUTION_ENDPOINT}/s3credentials`,
+          { followRedirect: false }
+        );
+        const authorizeUrl = new URL(response.headers.location);
+        expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+        expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      });
+
+      it('downloads a public science file', async () => {
+        accessToken = randomId('accessToken');
+        const s3SignedUrl = await invokeLambdaForS3SignedUrl(publicFileUrl, accessToken);
+        const parts = new URL(s3SignedUrl);
+        const userName = parts.searchParams.get('x-EarthdataLoginUsername');
+
+        const fileStream = got.stream(s3SignedUrl);
+        const downloadChecksum = await generateChecksumFromStream('cksum', fileStream);
+        expect(userName).toEqual('publicAccess');
+        expect(downloadChecksum).toEqual(fileChecksum);
+      });
+
+      it('refuses downloads of files in private buckets (by redirecting for authentication)', async () => {
+        accessToken = randomId('accessToken');
+        const notASignedUrl = await invokeLambdaForS3SignedUrl(privateFileUrl, accessToken);
+        const authorizeUrl = new URL(notASignedUrl);
+        expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+        expect(authorizeUrl.searchParams.get('state')).toEqual(`/${privateBucketName}/${fileKey}`);
+      });
     });
 
-    it('downloads the requested science file for authorized requests', async () => {
-      accessToken = await getTestAccessToken();
-      // Compare checksum of downloaded file with expected checksum.
-      const fileStream = await getDistributionApiFileStream(fileUrl, accessToken);
-      const downloadChecksum = await generateChecksumFromStream('cksum', fileStream);
-      expect(downloadChecksum).toEqual(fileChecksum);
+    describe('an authorized user', () => {
+      it('downloads a public science file', async () => {
+        accessToken = await getTestAccessToken();
```

Comment:
Any reason not to do this once in a `beforeAll` for this `describe` and share it across the tests?

Reply:
No reason. I'll do that